src/Plot/RobustPlot.purs

```diff
@@ -0,0 +1,111 @@
+module Plot.RobustPlot where
+
+import Prelude
+
+import Data.Array (fold, length, tail, zipWith, (!!), (..))
+import Data.Either (Either(..))
+import Data.Foldable (sum)
+import Data.Int (toNumber)
+import Data.Maybe (fromMaybe)
+import Data.Number (isFinite)
+import Data.Tuple (Tuple(..))
+import Draw.Actions (drawEnclosure)
+import Draw.Commands (DrawCommand)
+import Expression.Evaluator (evaluate)
+import Expression.Syntax (Expression)
+import IntervalArith.Approx (Approx, boundsNumber, fromRationalBoundsPrec)
+import IntervalArith.Misc (Rational, rationalToNumber, toRational)
+import Plot.Helper (drawLabel)
+import Types (XYBounds, Polygon, Position, Size)
+
+drawRobustPlot :: Size -> Int -> Int -> XYBounds -> Expression -> String -> DrawCommand Unit
+drawRobustPlot canvasSize numberOfPlots index bounds expression label = drawCommands
+  where
+  f = evaluateWithX expression
+
+  points = plotPoints canvasSize bounds f
+
+  labelPosition = toPosition $ fromMaybe [ { x: 0.0, y: 0.0 } ] $ points !! ((length points) / ((numberOfPlots + 1) * index))
+
+  drawCommands = fold [ drawPlot points, drawLabel label labelPosition ]
+
+evaluateWithX :: Expression -> Approx -> Approx
+evaluateWithX expression x = value
+  where
+  variableMap = [ Tuple "x" x ]
+
+  value = case evaluate variableMap expression of
+    Left _ -> zero -- TODO Handle any evaluation erros 
+    Right v -> v
+
+plotPoints :: Size -> XYBounds -> (Approx -> Approx) -> Array Polygon
+plotPoints canvasSize bounds f = points
+  where
+  segmentCount = 50
+
+  rangeX = bounds.xBounds.upper - bounds.xBounds.lower
+
+  rangeY = rationalToNumber $ bounds.yBounds.upper - bounds.yBounds.lower
+
+  domainSegments = map (toRational >>> toDomainX) $ 0 .. segmentCount
```

Comment:
Please rename:
domainSegments -> domainBreakpoints
defaultRange -> domainSegments
points -> segmentEnclosures
toCanvasPoint -> toCanvasEnclosure

Otherwise this looks very good.

Reply:
Resolved in the latest commit

Reply:
Thank you!